lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs

```diff
@@ -102,8 +102,28 @@ impl Client {
     /// It updates the local clipboard cache and returns the encoded message
     /// that should be sent to the RDP server.
     pub fn update_clipboard(&mut self, data: Vec<u8>) -> RdpResult<Vec<u8>> {
+        const CR: u8 = 0x13;
+        const LF: u8 = 0x10;
+
+        // convert LF to CRLF, as required by CF_OEMTEXT
+        let len_orig = data.len();
+        let mut converted = Vec::with_capacity(len_orig);
+        for i in 0..len_orig {
+            match data[i] {
+                LF if (i != 0) && data[i - 1] != CR => {
+                    converted.push(CR);
+                    converted.push(LF);
```

Comment:
Yes, vec will re-allocate a new block of memory and copy the data over as it needs to grow.

As pointed out by Jim, we could attempt to allocate with enough capacity to avoid the need to reallocate, but we have no idea how much space we'll need, and it doesn't really matter because this is not performance-critical code.

The original `Vec::with_capacity(len_orig)` at least gives us a good starting point, so chances are we will only need one re-allocation as opposed to several if we omitted that.